lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -20,10 +21,14 @@ defmodule SacaStatsWeb.CharacterController do
         |> redirect(to: Routes.character_path(conn, :character_search))
 
       {:ok, %PS2.API.QueryResult{data: body}} ->
+        next_rank = Map.get(body, "battle_rank")
+        {next_rank, ""} = Integer.parse(Map.get(next_rank, "value"))
+        next_rank = next_rank + 1
```

Comment:
```suggestion
        next_rank = 
          body
          |> get_in(["battle_rank", "value"])
          |> String.to_integer()
          |> Kernel.+(1)
```
One feature I love in Elixir is pipes (`|>`). The pipe operator takes the result of the expression on the left side (or in this case, on the line above), and uses it as the first argument in the function call on the right. E.g. `"26" |> String.to_integer()` is the same as `String.to_integer("26")`.

`get_in/2` is also a cool function - it makes it very easy to get nested values.

Reply:
Yeah I'm not the best with the elixir piping yet lol